models/articles.models.js

```diff
@@ -0,0 +1,12 @@
+const db = require("../db/connection.js");
+
+exports.selectArticleById = (id) => {
+  return db
+    .query("SELECT * FROM articles WHERE article_id = $1", [id])
+    .then((result) => {
+      if (result.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not found!" });
```

Comment:
Love the custom reject here, if you're going to do this, I would advise making the error message more descriptive to the client. What is not found? Personally I would change it to "Article not found" as that is what we are looking for with a valid id. This is optional as at this point this is just an opinion :)